src/powerhouse/components/tree-view-item/TreeViewItem.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+import CaretIcon from '../../../assets/icons/caret.svg';
+import VerticalDots from '../../../assets/icons/vertical-dots.svg';
+
+export interface TreeViewItemProps {
+    label: string;
+    children?: React.ReactNode;
+    initialOpen?: boolean;
+    expandedIcon?: string;
+    icon?: string;
+    level?: number;
+    onClick?: React.MouseEventHandler;
+    onOptionsClick?: React.MouseEventHandler;
+    secondaryIcon?: string;
+}
+
+const injectLevelProps = (
+    children: React.ReactNode,
+    level: number,
+): React.ReactNode => {
+    return React.Children.map(children, child => {
+        if (React.isValidElement(child)) {
+            if (child.type === React.Fragment) {
+                return injectLevelProps(child.props.children, level);
+            }
+
+            const customProps: Partial<TreeViewItemProps> = {};
+
+            if (child.type === TreeViewItem) {
+                customProps.level = level + 1;
+            }
+
+            return React.cloneElement(child, {
+                ...child.props,
+                ...customProps,
+            });
+        }
+
+        return child;
+    });
+};
+
+export const TreeViewItem: React.FC<TreeViewItemProps> = props => {
+    const {
+        icon,
+        label,
+        onClick,
+        children,
+        initialOpen,
+        expandedIcon,
+        secondaryIcon,
+        onOptionsClick,
+        level = 0,
+    } = props;
+
+    const [open, setOpen] = useState(initialOpen);
+
+    const toggleOpen = () => {
+        setOpen(!open);
+    };
+
+    useEffect(() => {
+        setOpen(initialOpen);
+    }, [initialOpen]);
+
+    const onClickItemHandler: React.MouseEventHandler = e => {
+        toggleOpen();
+        onClick && onClick(e);
+    };
+
+    const onOptionsClickHandler: React.MouseEventHandler = e => {
+        e.stopPropagation();
+        onOptionsClick && onOptionsClick(e);
+    };
+
+    return (
+        <div>
```

Comment:
TreeViewItemProps should extend React.HTMLAttributes<HTMLDivElement> and apply them at the root element so users of the component can set attributes like classNames.

Reply:
fixed in 5f772eefd6a8997a4f7e09e7c1a4f2f14ac2838a

Reply:
As a convention, let's merge the `className` prop with the default className, so someone can just add a new class without repeating all the default ones. The same for `style`.
```
onClick={onClickItemHandler}
style={{ paddingLeft: `${level * 10}px`, ...style }}
className={twMerge("flex flex-row cursor-pointer select-none rounded-lg py-3 group hover:bg-[#F1F5F9]", className)}
{...divProps}
```

Reply:
fixed in f4205e3fd37df68915423318c03fcf0b3b8ef0c9